documentation/src/docs/pravega-concepts.md

```diff
@@ -11,429 +11,312 @@ You may obtain a copy of the License at
+mechanism for state shared between multiple processes running in a cluster and making it easier to build distributed applications.  With State Synchronizer, an application developer can use Pravega to read and make changes to shared state consistently and perform optimistic locking. 
 
 ![State synchroner](img/state.synchronizer.png) 
 
 State Synchronizer could be used to maintain a single, shared copy of an
 application's configuration property across all instances of that application in
 a cloud.  State Synchronizer could also be used to store one piece of data or a
-map with thousands of different key value pairs.  In fact, Pravega itself uses
-State Synchronizer internally, to manage the state of ReaderGroups and Readers
-distributed throughout the network.
-
-An app developer creates a State Synchronizer on a Stream in a fashion similar
-to how s/he creates a Writer.  The State Synchronizer keeps a local copy
-of the shared state to make access to the data really fast for the application.
- Any changes to shared state are written through the StateSynchronizer to the
-Stream keeping track of all changes to the shared state.
- Each application instance uses the State Synchronizer to stay up to date with
-changes by pulling updates to shared state and modifying the local copy of the
-data.  Consistency is maintained through a conditional append style of updates
+map with thousands of different key value pairs. In Pravega, managing the state of ReaderGroups and distribution of Readers throughout the network is implemented using State Synchronizer.
+
+An application developer creates a State Synchronizer on a Stream similar to the creation of Writer. The State Synchronizer keeps a local copy of the shared state and allows faster access to the data in the application. State Synchronizer keeps track of all the changes happening in the shred state and responsible for performing any modification to the shared state in the Stream. Each application instance uses the State Synchronizer, to remain updated with the
+changes by pulling updates to the shared state and modifies the local copy of the
+data. Consistency is maintained through a conditional append style of updates
 to the shared state through the State Synchronizer, making sure that updates are
 made only to the most recent version of the shared state.
 
-The State Synchronizer can occasionally be "compacted", compressing and removing
-old state updates so that only the most recent version of the state is kept in
-the backing stream.  This feature helps app developers make sure that shared
-state does not grow unchecked.
-
- 
+The State Synchronizer can occasionally be "compacted", by compressing and removing
+older updates and retains only the recent version of the state in the backing Stream. This feature assures the application developers, that the shared state does not grow unchecked.
```

Comment:
Please, rephrase: `by compressing and removing older updates, while retaining only the most recent version of the state in the backing Stream`

Reply:
Accepted Suggestion